joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/datastructures/ProgramSummary.scala

```diff
@@ -299,6 +308,38 @@ trait TypeLike[M <: MethodLike, F <: FieldLike] {
     */
   def fields: List[F]
 
+  /** Adds the contents of the two types and produces a new type.
+    * @param o
+    *   the other type-like.
+    * @return
+    *   a type-like that is the combination of the two, with precedence to colliding contents to this type (LHS).
+    */
+  @targetName("add")
+  def +(o: TypeLike[M, F]): TypeLike[M, F]
+
+  /** Helper method for creating the sum of two type-like's methods, while preferring this types' methods on collisions.
+    * @param o
+    *   the other type-like.
+    * @return
+    *   the combination of the two type-like's methods.
+    */
+  protected def mergeMethods(o: TypeLike[M, _]): List[M] = {
```

Comment:
Can we have a test where a `builtin` type is used, but the same dependency is also present in the `.csproj` file?

Reply:
Example would be the `ModelBuilder` class in `Microsoft.EntityFrameworkCore`.